thirdeye-server/src/main/java/ai/startree/thirdeye/auth/AuthorizationManager.java

```diff
@@ -72,23 +82,41 @@ public void ensureCanValidate(final ThirdEyePrincipal principal, final AlertDTO
     ensureCanCreate(principal, entity);
   }
 
+  public <T extends AbstractDTO> void ensureHasAccess(final ThirdEyePrincipal principal,
+      final T entity, final AccessType accessType) {
+    ensureHasAccess(principal, dtoToId(entity), accessType);
+  }
+
   public void ensureHasAccess(final ThirdEyePrincipal principal,
       final ResourceIdentifier identifier, final AccessType accessType) {
     if (!hasAccess(principal, identifier, accessType)) {
       throw forbiddenExceptionFor(principal, identifier, accessType);
     }
   }
 
+  public <T extends AbstractDTO> boolean hasAccess(final ThirdEyePrincipal principal,
+      final T entity, final AccessType accessType) {
+    return hasAccess(principal, dtoToId(entity), accessType);
+  }
+
   public boolean hasAccess(final ThirdEyePrincipal principal,
       final ResourceIdentifier identifier, final AccessType accessType) {
-    return accessControl.hasAccess(principal, identifier, accessType);
+    return accessControl.hasAccess(principal.authToken, identifier, accessType);
+  }
+
+  static public <T extends AbstractDTO> ResourceIdentifier dtoToId(final T dto) {
```

Comment:
nice! I'd just call it `resourceIdentifier(entity)` or `resourceId(entity)`. `dtoToId` is a bit confusing since `id` is also the database id and doesn't call out resourceId